src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const historyClones = [];
+  const currentSt = { ...state };
+
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    switch (type) {
+      case 'clear':
+        for (const i in currentSt) {
+          delete currentSt[i];
+        }
```

Comment:
Instead of deleting properties one by one, you can simply assign an empty object to 'currentSt' to clear it. This is more efficient and easier to understand.